Common subdirectories: gaym-plugin-release-0.18/src/.deps and gaym-current/src/.deps
diff -u gaym-plugin-release-0.18/src/gaym.c gaym-current/src/gaym.c
--- gaym-plugin-release-0.18/src/gaym.c	2004-12-02 22:29:08.927084544 -0600
+++ gaym-current/src/gaym.c	2004-12-02 03:48:39.000000000 -0600
@@ -40,7 +40,7 @@
 
 #include "gaym.h"
 
-
+char *gaym_mask_bio(const char *biostring);
 static const char *gaym_blist_icon(GaimAccount *a, GaimBuddy *b);
 static void gaym_blist_emblems(GaimBuddy *b, char **se, char **sw, char **nw, char **ne);
 static GList *gaym_away_states(GaimConnection *gc);
@@ -165,25 +165,44 @@
 	struct gaym_conn *gaym=gc->proto_data;
 	GaimAccount* account = gaim_connection_get_account(gc);
 	char* hostname="none";
-	char* buf;
-	gaym->bio = g_strdup(info);
-	gaim_account_set_user_info(account, info);
-	gaim_account_set_string(account, "bioline",info);
-	buf = gaym_format(gaym, "vvvv:", "USER", gaim_account_get_username(account), hostname, gaym->server,
-			  gaym->bio);
+	char* buf, *bioline;
+
+	if (gaym->bio)
+		g_free (gaym->bio);
+		
+	if (info && strlen(info) > 2) {
+		gaym->bio = g_strdup(info);
+	} else if (gaym->server_bioline && strlen(gaym->server_bioline) > 2) {
+		gaym->bio = g_strdup(gaym_mask_bio(gaym->server_bioline));
+	} else {
+		gaym->bio = g_strdup("Gaim User");
+	}
+	
+	gaim_account_set_user_info(account, gaym->bio);
+	gaim_account_set_string(account, "bioline",gaym->bio);
+	gaim_debug_info ("gaym","INFO=%s BIO=%s\n",info,gaym->bio);
+	gaim_debug_misc("gaym","In login_cb, gc->account=%x\n",gc->account);
+	bioline=g_strdup_printf("%s#%s", gaym->thumbnail,gaym->bio); 
+
+	buf = gaym_format(gaym, "vvvv:", "USER", 
+		gaim_account_get_username(account), 
+		hostname, gaym->server, bioline);
+	
+	g_free (bioline);
 	
 	if (gaym_send(gaym, buf) < 0) {
 		gaim_connection_error(gc, "Error registering with server");
 		return;
 	}
 }
+
 static void gaym_show_set_info(GaimPluginAction *action)
 {
 	GaimConnection *gc = (GaimConnection *) action->context;
-	gaim_debug_misc("gaym","in show-set_info, gc->account=%x\n",gc->account);
-	
+	gaim_debug_misc("gaym","in show-set_info, gc->account=%x\n",gc->account);	
 	gaim_account_request_change_user_info(gaim_connection_get_account(gc));
 }
+
 static GList *gaym_actions(GaimPlugin *plugin, gpointer context)
 {
 	GList *list = NULL;
@@ -471,11 +490,6 @@
     last_room=room;
     }
     
-   
-    
-    
-       
-        
 }
 
 static void gaym_login_cb(gpointer data, gint source, GaimInputCondition cond)
@@ -518,6 +532,7 @@
 	hostname[sizeof(hostname) - 1] = '\0';
 	username = gaim_account_get_string(gaym->account, "username", "");
 	user_bioline = gaim_account_get_string(gaym->account, "bioline", ""); 
+	gaim_debug_info ("gaym","USER BIOLINE=%s\n",user_bioline);
 	gaim_account_set_user_info(gc->account, user_bioline);
 	gaim_debug_misc("gaym","In login_cb, gc->account=%x\n",gc->account);
 	bioline=g_strdup_printf("%s#%s", gaym->thumbnail,user_bioline); 
@@ -592,8 +607,6 @@
 	args[0] = who;
 	gaym->info_window_needed=TRUE;
 	gaym_cmd_whois(gaym, "whois", NULL, args);
-	
-	
 }
 
 static void gaym_set_away(GaimConnection *gc, const char *state, const char *msg)
diff -u gaym-plugin-release-0.18/src/msgs.c gaym-current/src/msgs.c
--- gaym-plugin-release-0.18/src/msgs.c	2004-12-02 22:29:08.921085480 -0600
+++ gaym-current/src/msgs.c	2004-12-02 02:28:23.000000000 -0600
@@ -39,7 +39,7 @@
 static void gaym_chat_remove_buddy(GaimConversation *convo, char *data[2]);
 static void gaym_buddy_status(char *name, struct gaym_buddy *ib, struct gaym_conn *gaym);
 
-static char * gaym_mask_thumbnail(const char* biostring)
+char * gaym_mask_thumbnail(const char* biostring)
 {    
   char * start = strchr(biostring,':');
   char * end=0;
@@ -59,13 +59,14 @@
 }
 
 
-char* gaym_bot_detect(char * bio,GaimConversation *chat,const char *name)
+char * gaym_bot_detect(char * bio,GaimConversation *chat,const char *name)
 {
 	static char * appendStr = 
 		"  <b><font size=2>** BOT ALERT **</font></b>";
 	static char *badlines[] = {
                 "geocities","tripod","angelcities",
 		"icamsonline","ratefun","dudepages",
+		"h0rnydolls","hornydolls",
                 "gayonlinevideos","gaystreamingvideos",
                 "gaypornfilms",
 		NULL
@@ -144,7 +145,7 @@
 	return bio;
 }
 
-static char* gaym_mask_bio(const char* biostring)
+char* gaym_mask_bio(const char* biostring)
 {
   char * start = strchr(biostring,'#');
   char* end=0;
diff -u gaym-plugin-release-0.18/src/weblogin.c gaym-current/src/weblogin.c
--- gaym-plugin-release-0.18/src/weblogin.c	2004-11-24 04:42:40.000000000 -0600
+++ gaym-current/src/weblogin.c	2004-12-02 03:49:10.000000000 -0600
@@ -63,6 +63,22 @@
   GaimUrlSession* session;
 
 } GaimFetchUrlData;
+
+/* gaim_url_decode doesn't change pluses to spaces - edit in place */
+static const char * gaym_url_decode (const char *string)
+{
+	char * retval;
+
+	gaim_debug_info ("gaym","Changing %s with gaym_url_decode\n", string);
+	retval = string = gaim_url_decode(string);
+	while (*retval != NULL) {
+		if (*retval == '+')
+			*retval = ' ';
+		retval++;
+	}
+	return string;
+}
+
 static void
     destroy_fetch_url_data(GaimFetchUrlData *gfud)
 {
@@ -496,7 +512,7 @@
 	if(temp && temp2)
 	{
 		thumbnail=g_strndup(temp,(temp2-temp)*sizeof(char));
-		result=gaim_url_decode(thumbnail);
+		result=gaym_url_decode(thumbnail);
 		(gaym->thumbnail=g_strdup(result)) ||(gaym->thumbnail=g_strdup(" "));
 		
 		g_free(thumbnail);
@@ -505,17 +521,16 @@
 		if(temp) 
 		{
 			bio=g_strndup(temp2,(temp-temp2)*sizeof(char));
-			result=gaim_url_decode(bio);
+			result=gaym_url_decode(bio);
+			gaim_debug_info("gaym","Server BIO: %s Thumb: %s\n",
+				result, gaym->thumbnail);
 			(gaym->server_bioline=g_strdup(result))
-					||(gaym->server_bioline=g_strdup(" "));
+					||(gaym->server_bioline=NULL);
 			g_free(bio);
 			
 		}
 		//We have established a session. Call session callback.
 		
-		
-	
-		
 	}
 	else
 	{
